app/presenters/date_question_presenter.rb

```diff
@@ -9,17 +10,54 @@ def response_label(value)
     end
   end
 
-  def start_date
-    @node.range == false ? 1.year.ago : @node.range.begin
+  def error_id
+    "error_id" if error
   end
 
-  def end_date
-    @node.range == false ? 3.years.from_now : @node.range.end
+  def days_options
+    days = Array(1..31).map { |number|
+      {
+        text: number,
+        value: number,
+        selected: default_for_date(prefill_value_for(self, :day)) == number,
+      }
+    }
+    days.unshift(text: "", value: "")
+  end
+
+  def months_options
+    months = Array(1..12).map { |number|
+      {
```

Comment:
I wonder if we can simplify some of this logic into one method, as there's a lot of repetition for each days, months, years method. Something like this, which can be called by each method?

```
def format_date(number, type)
    {
      text: type.eql?(:month) ? Date::MONTHNAMES[number] : number,
      value: number,
      selected: default_for_date(prefill_value_for(self, type)) == number,
    }
  end
```

Reply:
Nice. Thanks for that. Have implemented.

Reply:
Have also fixed the commit messages, I had cleaned up some of them, but obviously had missed a couple.